### PR TITLE
5579 fix remaining report bugs

### LINKED ISSUE #5579: Item usage report showing different data on mSupply vs on OMS
## What went wrong? 😲
Numbers are different on both reports. Would expect numbers to be the same. Am on latest version of mSupply and OMS.

As an example, see item code: PMC834

Report comparisons:
mSupply:
Screenshot 2024-11-27 at 10 58 06 AM

OMS:
Screenshot 2024-11-27 at 10 58 24 AM

This is the stocklines on OMS and Legacy:
Screenshot 2024-11-27 at 11 08 07 AM

Screenshot 2024-11-27 at 11 08 50 AM

In stock has 1 extra item, stock on order is 0, expiring months is 0. 

## Expected behaviour 🤔
All the values of mSupply report and OMS report should match

## How to Reproduce 🔨

Steps to reproduce the behaviour:
Generate a report on mSupply and on OMS and compare

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

### PR DESCRIPTION
**Pull request by fergie-nz**
Fixes #5579 

# 👩🏻‍💻 What does this PR do?

- Adds AMC column to expiring items report
- Prevents negative numbers for stock at risk on expiring items report
- Prevents negative numbers for stock on order in item usage report
- Update item usage in stock to be available stock, not total stock. Also update column header
- Update test suite

## 💌 Any notes for the reviewer?

You can test this by deleting old reports in database, then running:

```
cargo build &&  ./target/debug/remote_server_cli build-standard-reports && ./target/debug/remote_server_cli upsert-reports-json
```

# 🧪 Testing

- [ ] Rebuild and upsert reports with above command (first delete old reports from database)
- [ ] open item usage report - see changed field header and data for available stock
- [ ] See stock on order shows 0 where previously showed negative numbers
- [ ] open expiring items report - see added AMC column
- [ ] See stock at risk shows 0 where previously showed negative numbers

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
Later will need to update documentation to explain how fields are calculated in OMS reports.

### CODE REVIEW

File: server/reports/expiring-items/2_3_0/convert_data_js/src/utils.js [hunk 1]
```diff
@@ -77,6 +80,9 @@ const calculateStockAtRisk = (
       }
     }
   }
+  if (stockAtRisk < 0) {
```
Comment (andreievg) [inline server/reports/expiring-items/2_3_0/convert_data_js/src/utils.js:83]: I think expected usage should also be capped

Comment (andreievg) [reply server/reports/expiring-items/2_3_0/convert_data_js/src/utils.js:83]: https://github.com/msupply-foundation/open-msupply/compare/5579-fix-remaining-report-bugs...5579-fix-remaining-report-bugs(Cap-expected-usage-to-totalNumberOfPacks)

**Review by andreievg — APPROVED** [review]
Looks great thanks Ferg ! One small change before you merge (as per suggestion), it's the expected usage should not be higher then soh

**Comment by fergie-nz** [comment]
> Looks great thanks Ferg ! One small change before you merge (as per suggestion), it's the expected usage should not be higher then soh

Nice! Sounds good, thanks Andrei

**Pull request merged**